DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/info/StringAnalysisReflectiveCalls.scala

```diff
@@ -0,0 +1,311 @@
+        //"javax.crypto.CipherSpi#engineUnwrap", "javax.crypto.EncryptedPrivateKeyInfo#getKeySpec",
+        //"javax.crypto.ExemptionMechanism#getInstance", "javax.crypto.KeyAgreement#getInstance",
+        //"javax.crypto.KeyGenerator#getInstance", "javax.crypto.Mac#getInstance",
+        //"javax.crypto.SealedObject#getObject", "javax.crypto.SecretKeyFactory#getInstance"
+        // The following is for the Java Reflection API
+        "java.lang.Class#forName", "java.lang.ClassLoader#loadClass",
+        "java.lang.Class#getField", "java.lang.Class#getDeclaredField",
+        "java.lang.Class#getMethod", "java.lang.Class#getDeclaredMethod"
+    )
+
+    /**
+     * A list of fully-qualified method names that are to be skipped, e.g., because they make an
+     * analysis crash (e.g., com/sun/jmx/mbeanserver/MBeanInstantiator#deserialize)
+     */
+    private val ignoreMethods = List()
+
+    // executeFrom specifies the index / counter when to start feeding entities to the property
+    // store. executeTo specifies the index / counter when to stop feeding entities to the property
+    // store. These values are basically to help debugging. executionCounter is a helper variable
+    // for that purpose
+    private val executeFrom = 0
+    private val executeTo = 10000
+    private var executionCounter = 0
+
+    override def title: String = "String Analysis for Reflective Calls"
+
+    override def description: String = {
+        "Finds calls to methods provided by the Java Reflection API and tries to resolve passed "+
+            "string values"
+    }
+
+    /**
+     * Using a `declaringClass` and a `methodName`, this function returns a formatted version of the
+     * fully-qualified method name, in the format [fully-qualified class name]#[method name]
+     * where the separator for the fq class names is a dot, e.g., "java.lang.Class#forName".
+     */
+    private def buildFQMethodName(declaringClass: ReferenceType, methodName: String): String =
+        s"${declaringClass.toJava}#$methodName"
+
+    /**
+     * Taking the `declaringClass` and the `methodName` into consideration, this function checks
+     * whether a method is relevant for this analysis.
+     *
+     * @note Internally, this method makes use of [[relevantMethodNames]]. A method can only be
+     *       relevant if it occurs in [[relevantMethodNames]].
+     */
+    private def isRelevantCall(declaringClass: ReferenceType, methodName: String): Boolean =
+        relevantMethodNames.contains(buildFQMethodName(declaringClass, methodName))
+
+    /**
+     * Helper function that checks whether an array of [[Instruction]]s contains at least one
+     * relevant method that is to be processed by `doAnalyze`.
+     */
+    private def instructionsContainRelevantMethod(instructions: Array[Instruction]): Boolean = {
+        instructions.filter(_ != null).foldLeft(false) { (previous, nextInstr) ⇒
```

Comment:
The foldLeft does not make use of possible early returns (e.g. long list of instructions, and the first one is relevant).